File: articles/forecasting-and-decision-support/r/run_all_forecasting_workflows.R
# run_all_forecasting_workflows.R
# Run all R workflows for Forecasting and Decision Support.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "forecasting_decision_support_workflow.R",
  "forecast_accuracy_profiles.R",
  "calibration_reliability_tables.R",
  "threshold_decision_tables.R",
  "forecast_value_reports.R",
  "horizon_degradation_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R forecasting workflows completed.")

File: articles/forecasting-and-decision-support/python/run_all_forecasting_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Forecasting and Decision Support."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "forecast_error_metrics.py",
    "probabilistic_forecast_calibration.py",
    "decision_threshold_forecasts.py",
    "forecast_value_of_information.py",
    "horizon_degradation_analysis.py",
    "reference_class_forecasting.py",
    "early_warning_signal_monitoring.py",
    "decision_record_exporter.py",
    "forecasting_decision_support_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python forecasting workflows completed.")


if __name__ == "__main__":
    main()
